src/student/customImplementations/DynamicArray.java

```diff
@@ -0,0 +1,67 @@
+package student.customImplementations;
+
+import student.models.Student;
+
+public class DynamicArray {
+    private Student[] sArray;
+
+    public void add(Student student) {
+        if (sArray == null) {
+            sArray = new Student[1];
+            sArray[0] = student;
+        } else {
+            Student[] curArray = new Student[sArray.length + 1];
```

Comment:
in your implementation in all cases when we add some new element in dynamic array it will resize it, which is not optimal, the optimal and high-performance way is to create it with some capacity and increase the size when it reaches to that capacity, check JDK implementation of ArrayList, or contact me :) if you don't understand how do that Nelli jan